azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -141,11 +254,8 @@ public long getLastCleanerThreadCheckTime() {
   public Set<String> getAllActiveExecutorServerHosts() {
     // Includes non primary server/hosts
     HashSet<String> ports = new HashSet<String>();
-    ports.add(executorHost + ":" + executorPort);
-    for (Pair<ExecutionReference, ExecutableFlow> running : runningFlows
-        .values()) {
-      ExecutionReference ref = running.getFirst();
-      ports.add(ref.getHost() + ":" + ref.getPort());
+    for (Executor executor : activeExecutors) {
```

Comment:
Are we saying activeExecutors can contain duplicate server hosts and that is why ports is of type HashSet?